c_common/front_end_common_lib/include/spinn_extra.h

```diff
@@ -0,0 +1,1106 @@
+
+#ifndef __SPINN_EXTRA_H__
+#define __SPINN_EXTRA_H__
+
+#include <spinnaker.h>
+#include <stdbool.h>
+#if defined(__GNUC__) && __GNUC__ < 6
+// This particular warning (included in -Wextra) is retarded wrong for client
+// code of this file. Only really a problem on Travis.
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+#endif // __GNUC__
+
+// Generates valid code if the named type is one word long, and invalid code
+// otherwise. It would be simpler if we used _Static_assert, but Eclipse seems
+// to hate that.
+#define ASSERT_WORD_SIZED(type_ident) \
+    struct __static_word_sized_assert_ ## type_ident { \
+        char _dummy[2 * (sizeof(type_ident) == sizeof(uint)) - 1]; \
+    }
+
+// ---------------------------------------------------------------------
+// 1. Chip Organization
+
+// No registers
+
+// ---------------------------------------------------------------------
+// 2. System Architecture
+
+// No registers
+
+// ---------------------------------------------------------------------
+// 3. ARM968 Processing Subsystem
+
+// No registers
+
+// ---------------------------------------------------------------------
+// 4. ARM 968
+
+// No special registers here
+
+// ---------------------------------------------------------------------
+// 5. Vectored Interrupt Controller
+
+typedef void (*vic_interrupt_handler_t) (void);
+
+typedef union {
+    struct {
+        uint watchdog : 1;
+        uint software : 1;
+        uint comm_rx : 1;
+        uint comm_tx : 1;
+        uint timer1 : 1;
+        uint timer2 : 1;
+        uint cc_rx_ready : 1;
+        uint cc_rx_parity_error : 1;
+        uint cc_rx_framing_error : 1;
+        uint cc_tx_full : 1;
+        uint cc_tx_overflow : 1;
+        uint cc_tx_empty : 1;
+        uint dma_done : 1;
+        uint dma_error : 1;
+        uint dma_timeout : 1;
+        uint router_diagnostic : 1;
+        uint router_dump : 1;
+        uint router_error : 1;
+        uint cpu : 1;
+        uint ethernet_tx : 1;
+        uint ethernet_rx : 1;
+        uint ethernet_phy : 1;
+        uint slow_clock : 1;
+        uint cc_tx_not_full : 1;
+        uint cc_rx_mc : 1;
+        uint cc_rx_p2p : 1;
+        uint cc_rx_nn : 1;
+        uint cc_rx_fr : 1;
+        uint int0 : 1;
+        uint int1 : 1;
+        uint gpio8 : 1;
+        uint gpio9 : 1;
+    };
+    uint value;
+} vic_mask_t;
+
+typedef struct {
+    const vic_mask_t irq_status;
+    const vic_mask_t fiq_status;
+    const vic_mask_t raw_status;
+    vic_mask_t int_select;
+    vic_mask_t int_enable;
+    vic_mask_t int_disable;
+    vic_mask_t soft_int_enable;
+    vic_mask_t soft_int_disable;
+    bool protection;
+    const uint _padding[3];
+    vic_interrupt_handler_t vector_address;
+    vic_interrupt_handler_t default_vector_address;
+} vic_control_t;
+
+typedef struct {
+    uint source : 5;
+    uint enable : 1;
+    uint : 26;
+} vic_vector_control_t;
+
+ASSERT_WORD_SIZED(vic_mask_t);
```

Comment:
comment saying we're ensuring structs are word aligned?

Reply:
The macro itself is documented as doing that. And it's not _alignment_ but _sizing_.